packages/jsdoc-plugins/lib/relation-fixer/addmissingdoclets.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license Copyright (c) 2017, CKSource - Frederico Knabben. All rights reserved.
+ * Licensed under the terms of the MIT License (see LICENSE.md).
+ */
+
+'use strict';
+
+const isNonEmptyArray = require( './utils/isnonemptyarray' );
+const cloneDeep = require( 'lodash' ).cloneDeep;
+
+/**
+ * Adds missing doclets of static members coming from implemented interfaces and extended classes.
+ * JSDoc does not support inheritance of static members.
+ * This module requires the input to be processed by 'relationbuilder' first.
+ *
+ * @param {Array.<Doclet>} data
+ * @returns {Array.<Doclet>} data
+ */
+module.exports = data => {
```

Comment:
This function is overcomplicated, it contains many nested functions, scopes and some shadow variables, e.g. `newDoclets`.